apps/docs/pages/guides/auth/auth-postgres-hooks.mdx

```diff
@@ -0,0 +1,851 @@
+import Layout from '~/layouts/DefaultGuideLayout'
+import {CodeBlock, Admonition, Accordion} from 'ui'
+import Image from 'next/image'
+
+export const meta = {
+  id: 'auth-postgres-hooks',
+  title: 'Auth Hooks (Postgres)',
+  description: 'Use Supabase Postgres Functions to customize your authentication flow',
+}
+
+<Admonition type="info">
+Hooks are in beta. They will be available for all projects in early 2024.
+</Admonition>
+
+
+Supabase allows you to use PostgreSQL functions to alter the default Supabase Auth flow. Developers can use hooks to add custom behavior that's not supported natively.
+
+Hooks help you:
+
+- Track the origin of user signups by adding metadata
+- Improve security by adding additional checks to password and multi-factor authentication
+- Support legacy systems by integrating with identity credentials from external authentication systems
+- Add additional custom claims to your JWT
+
+
+You can use hooks at specific points along an Auth flow to perform custom behavior via Postgres Functions. These are known as **extensibility points**. The following extensibility points are available:
+
+| Extensibility Point              | Available on Plan       |
+|----------------------------------|-------------------------|
+| MFA Verification Attempt         | Teams and Enterprise    |
+| Password Verification Attempt    | Teams and Enterprise    |
+| Custom Access Token              | Free                    |
+
+You can link a hook to an extensiblity point to signal to Supabase Auth that it should make use of the hook.
+
+
+Below is a flow chart illustrating how Supabase Auth uses an Auth Hook:
+
+<Image
+  alt="Auth Hooks Flowchart"
+  width={'1000'}
+  height={'400'}
+  src="/docs/img/guides/auth-postgres-hooks/hooks-flowchart.png"/>
+
+
+## Create a Hook
+### Write a Hook
```

Comment:
I think this section is a bit too simplistic. 

Yes, you can mention the template (and less prominently), but the main goal of it should be to explain:

- What does an Auth hook function look like?
- That you can / cannot use JSON/JSONB.
- Link to Postgres docs about JSON/JSONB operators.
- Link to Postgres docs about functions and languages.
- Explain the roles and grants needed for a function.

Reply:
Sounds good, will include. The initial intention was that by using a template one could sidestep concerns like:

> What does an Auth hook function look like?
> That you can / cannot use JSON/JSONB.

Since the dev can edit the template and not have to worry about what it looks like.

> Link to Postgres docs about JSON/JSONB operators.
> Link to Postgres docs about functions and languages.

Thanks, it's a good suggestion. Let's add that.